src/widgets/dnshole/DnsHoleControls.tsx

```diff
@@ -124,53 +134,95 @@ function DnsHoleControlsWidgetTile({ widget }: DnsHoleControlsWidgetProps) {
     return dnsList;
   };
 
-  const toggleDns = async (action: 'enable' | 'disable', appsToChange?: string[]) => {
+  const toggleDns = async (
+    action: 'enable' | 'disable',
+    appsToChange?: string[],
+    hours: number = 0,
+    minutes: number = 0
+  ) => {
+    const duration = hours * 3600 + minutes * 60;
     await mutateAsync(
       {
         action,
+        duration,
         configName,
         appsToChange,
       },
       {
         onSettled: () => {
           trpcUtils.dnsHole.summary.invalidate();
+
+          setTimeout(
```

Comment:
it seems to be the old classic staleTime vs refetchTime.
Might be better to change useDnsHoleSummeryQuery's staleTime to refetchTime in DnsHoleSummary.tsx

I do understand that this timer specifically refetches after the set amount of time, but it might not be necessary if the refetchTime is set correctly.